packages/@aws-cdk/aws-ecs/lib/ecs-queue-worker-service.ts

```diff
@@ -0,0 +1,68 @@
+import cdk = require('@aws-cdk/cdk');
+import { Ec2Service } from './ec2/ec2-service';
+import { Ec2TaskDefinition } from './ec2/ec2-task-definition';
+import { QueueWorkerServiceBase, QueueWorkerServiceBaseProps } from './queue-worker-service-base';
+
+/**
+ * Properties to define an Ec2 query worker service
+ */
+export interface Ec2QueueWorkerServiceProps extends QueueWorkerServiceBaseProps {
+  /**
+   * The minimum number of CPU units to reserve for the container.
+   *
+   * @default none
+   */
+  readonly cpu?: number;
+
+  /**
+   * The hard limit (in MiB) of memory to present to the container.
+   *
+   * If your container attempts to exceed the allocated memory, the container
+   * is terminated.
+   *
+   * At least one of memoryLimitMiB and memoryReservationMiB is required for non-Fargate services.
+   */
+  readonly memoryLimitMiB?: number;
+
+  /**
+   * The soft limit (in MiB) of memory to reserve for the container.
+   *
+   * When system memory is under contention, Docker attempts to keep the
+   * container memory within the limit. If the container requires more memory,
+   * it can consume up to the value specified by the Memory property or all of
+   * the available memory on the container instance—whichever comes first.
+   *
+   * At least one of memoryLimitMiB and memoryReservationMiB is required for non-Fargate services.
+   */
+  readonly memoryReservationMiB?: number;
+}
+
+/**
+ * Class to create an Ec2 query worker service
+ */
+export class Ec2QueueWorkerService extends QueueWorkerServiceBase {
+  constructor(scope: cdk.Construct, id: string, props: Ec2QueueWorkerServiceProps) {
+    super(scope, id, props);
+
+    // Create a Task Definition for the container to start
+    const taskDefinition = new Ec2TaskDefinition(this, 'QueueWorkerTaskDef');
+    taskDefinition.addContainer('QueueWorkerContainer', {
+      image: props.image,
+      memoryLimitMiB: props.memoryLimitMiB,
+      memoryReservationMiB: props.memoryReservationMiB,
+      cpu: props.cpu,
+      command: props.command !== undefined ? cdk.Fn.split(",", props.command) : undefined,
+      environment: this.environment,
```

Comment:
Tests were added to expect specific environment and logdriver properties

Reply:
Oh, I had assumed you made `environment` an object member (as opposed to a local variable), because it could be mutated after construction, but I now see it's because the logic to determine the value is shared in the inherited constructor.

Hmm, not a fan of this pattern, but I also don't have a strongly better suggestion.

Disregard.